tests/kuttl/exoscale-status-test/00-install-exoscale-composition.yaml

```diff
@@ -0,0 +1,313 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    argocd.argoproj.io/sync-wave: '10'
+  labels:
+    name: xobjectbuckets.appcat.vshn.io
+  name: xobjectbuckets.appcat.vshn.io
+spec:
+  claimNames:
+    kind: ObjectBucket
+    plural: objectbuckets
```

Comment:
This feels like a manually maintained duplication from the golden tests of the packages.
Any thoughts about automating/generating it from the golden tests?
I'd rather not maintain 2 versions of a composition, as the test would be worthless since it's not testing the result of a commodore package.

Reply:
I thought about it and even started to work into that direction but I realised that those are 2 different kind of tests and they can and will deviate over time.  What I could do is to compile the package and copy the result but again I have to change namespaces or add fields such as annotations. I would really like to leave them separate or at worst to get them from compile and adjust the fields.

Reply:
> 2 different kind of tests and they can and will deviate over time

How so? What do you expect to change that is different from a package output vs the test case?

> to get them from compile and adjust the fields.

Then please do this, e.g. somewhere in a make target, using `yq` or so. But having them separate is a no-go IMO. We add the integration test for the package, because that's the thing that users change. It's too easy to forgot about adjusting the test case when users change configuration in a package (e.g. add a patch in composition) and that creates a false security.

Reply:
Actually I could argue to remove everything from the spec object and test only the conditions, in that case we do not have to maintain neither the kuttl install file nor the kuttl assert file. With this approach we do not have to maintain the test at all, we will test the conditions that they are propagated accordingly.

Reply:
> Updated: In any case I think I need to generate the composition and composite to make sure any changes to the composite and composition are reflected in the install step.

Yes, that's what I meant the whole time :slightly_smiling_face:

Reply:
Now they are autogenerated.